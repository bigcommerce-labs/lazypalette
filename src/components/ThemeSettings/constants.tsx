export const DebounceTimeMs = 500;

export enum SettingType {
    COLOR = 'color',
    CHECKBOX = 'checkbox',
    FONT = 'font',
    HEADING = 'heading',
    IMAGE_DIMENSION = 'imageDimension',
    INPUT = 'input',
    OPTIMIZED_CHECKOUT_IMAGE = 'optimizedCheckout-image',
    PARAGRAPH = 'paragraph',
    SELECT = 'select',
    TEXT = 'text',
}
