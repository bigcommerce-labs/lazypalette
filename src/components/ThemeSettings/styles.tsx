import { theme } from 'pattern-lab';
import styled from 'styled-components';

export const List = styled.ul`
    width: 20rem;
    max-height: 55vh;
    list-style: none;
    padding: 0;
    margin: 0 1rem 0 1.5rem;

    > :first-child > * {
        margin-top: 0;
    }
`;

export const Item = styled.li`
    line-height: 1.5rem;
    margin: 0 0 .5rem 0;
    max-width: 20rem;
`;

export const Heading = styled.div`
    color: ${theme.colors.primaryText};
    font-weight: ${theme.typography.fontWeight.bold};
    line-height: 1.5rem;
    margin: 2.5rem 0 0.5rem 0;
`;
