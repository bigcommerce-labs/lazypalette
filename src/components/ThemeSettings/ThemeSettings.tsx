import { CheckboxInput, InputField, SelectBox } from 'pattern-lab';
import React, { ChangeEvent, Component } from 'react';
import { connect } from 'react-redux';
import { withRouter, Route, RouteComponentProps } from 'react-router-dom';
import { Dispatch } from 'redux';

import { updateThemeConfigChange, SettingsType, ThemeConfigChange } from '../../actions/theme';
import { State } from '../../reducers/reducers';
import { ThemeSchemaEntry, ThemeSchemaEntrySetting } from '../../reducers/theme';
import CheckoutImageUpload from '../CheckoutImageUpload/CheckoutImageUpload';
import ColorPicker from '../ColorPicker/ColorPicker';
import ExpandableMenu from '../ExpandableMenu/ExpandableMenu';
import ImageSize from '../ImageSize/ImageSize';

import { Heading, Item, List } from './styles';

export interface ThemeSettingsProps extends RouteComponentProps<{}> {
    settings: SettingsType;
    settingsIndex: number;
    themeSettings: ThemeSchemaEntry;
    updateThemeConfigChange(configChange: ThemeConfigChange): (dispatch: Dispatch<State>) => void;
}

function transformOptions(setting: ThemeSchemaEntrySetting) {
    return setting.options ? setting.options.map(({label, value}) => ({
        label,
        value: value.toString(),
    })) : [];
}

function getEditor(
    setting: ThemeSchemaEntrySetting,
    preSetValue: SettingsType,
    handleChange: (event: ChangeEvent<HTMLInputElement | HTMLSelectElement>) => void,
    broadcastConfigChange: (configChange: ThemeConfigChange) => (dispatch: Dispatch<State>) => void
) {
    switch (setting.type) {
        case 'color':
            return <ColorPicker
                color={preSetValue[`${setting.id}`] as string}
                label={setting.label}
                onChange={handleChange}
            />;
        case 'checkbox':
            return <CheckboxInput
                checked={preSetValue[`${setting.id}`] as boolean}
                label={setting.label}
                onChange={handleChange}
            />;
        case 'font':
            return <SelectBox
                selected={preSetValue[`${setting.id}`] as string}
                label={setting.label}
                onChange={handleChange}
                options={transformOptions(setting)}
            />;
        case 'imageDimension':
            return <ImageSize
                selected={preSetValue[`${setting.id}`] as string}
                label={setting.label || ''}
                onChange={handleChange}
                options={transformOptions(setting)}
            />;
        case 'optimizedCheckout-image':
            return <CheckoutImageUpload
                label={setting.label || ''}
                name={setting.id!}
                onChange={broadcastConfigChange}
                imageURL={preSetValue[`${setting.id}`] as string}
            />;
        case 'select':
            return <SelectBox
                selected={preSetValue[`${setting.id}`] as string}
                label={setting.label}
                onChange={handleChange}
                options={transformOptions(setting)}
            />;
        case 'text':
            return <InputField
                value={preSetValue[`${setting.id}`] as string}
                label={setting.label}
                onChange={handleChange}
            />;
        case 'heading':
            return <Heading>{setting.content}</Heading>;
        default:
            return null;
    }
}

export class ThemeSettings extends Component<ThemeSettingsProps, {}> {
    handleChange = (setting: ThemeSchemaEntrySetting) =>
        ({target}: ChangeEvent<HTMLInputElement | HTMLSelectElement>) => {
            const value = target.type === 'checkbox' ? (target as HTMLInputElement).checked : target.value;

            this.props.updateThemeConfigChange({ setting, value });
        };

    render() {
        const { match, settings, settingsIndex, themeSettings } = this.props;

        return (
            <Route
                path={`/section/${settingsIndex}`}
                exact
                render={() => (
                    <ExpandableMenu
                        title={themeSettings ? themeSettings.name : ''}
                        back={match.url}>
                        <List>
                            {themeSettings.settings.map((setting, index) => (
                                <Item key={index}>
                                    {
                                        getEditor(
                                            setting,
                                            settings,
                                            this.handleChange(setting),
                                            this.props.updateThemeConfigChange
                                        )
                                    }
                                </Item>
                            ))}
                        </List>
                    </ExpandableMenu>
                )}
            />
        );
    }
}

const mapStateToProps = (state: State, ownProps: ThemeSettingsProps) => ({
    settings: state.theme.settings,
    themeSettings: state.theme.schema[ownProps.settingsIndex],
});

interface StateFromProps {
    settings: SettingsType;
    themeSettings: ThemeSchemaEntry;
}

interface ActionFromProps {
    updateThemeConfigChange(configChange: ThemeConfigChange): (dispatch: Dispatch<State>) => void;
}

const mapDispatchToProps = {
    updateThemeConfigChange,
};

export default withRouter(
    connect<StateFromProps, ActionFromProps, { settingsIndex: number }>(mapStateToProps,
        mapDispatchToProps)(ThemeSettings));
