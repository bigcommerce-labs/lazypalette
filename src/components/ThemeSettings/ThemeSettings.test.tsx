import {  shallow } from 'enzyme';
import React from 'react';

import { getEditor, ThemeSettings } from './ThemeSettings';

const settings = {
    brandpage_products_per_page: 12,
    categorypage_products_per_page: 12,
    homepage_blog_posts_count: 3,
    homepage_featured_products_column_count: 4,
    homepage_featured_products_count: 8,
    homepage_new_products_column_count: 4,
    homepage_new_products_count: 12,
    homepage_show_carousel: true,
    homepage_top_products_column_count: 4,
    homepage_top_products_count: 8,
    productpage_related_products_count: 10,
    productpage_reviews_count: 9,
    productpage_similar_by_views_count: 10,
    productpage_videos_count: 8,
    searchpage_products_per_page: 12,
    show_product_quick_view: true,
};

const preSetValue = {
    'test-imageDimension': '20x20',
    'test-imageUpload': 'hello_there',
    'test-select': 1,
    'test-text': 'kermit_the_frog',
    'testing-checkbox': false,
    'testing-color-picker': '#FFFFF',
    'testing-font': false,
};

const mockProp: any = jest.fn();
const routeProps = {
    history: mockProp,
    location: mockProp,
    match: mockProp,
    staticContext: mockProp,
};

describe('<ThemeSettings />', () => {
    describe('render()', () => {
        it('renders the component', () => {
            const mockFetch = jest.fn();
            const testItems = {
                name: 'Typography & Icons',
                settings: [
                    {
                        content: 'Logo',
                        type: 'heading',
                    },
                    {
                        id: 'logo-font',
                        label: 'Font family',
                        options: [
                            {
                                group: 'Oswald',
                                label: 'Oswald',
                                value: 'Google_Oswald_400',
                            },
                            {
                                group: 'Oswald',
                                label: 'Oswald Light',
                                value: 'Google_Oswald_300',
                            },
                        ],
                        type: 'font',
                    },
                    {
                        content: 'Headings',
                        type: 'heading',
                    },
                    {
                        id: 'color-textHeading',
                        label: 'Headings text color',
                        type: 'color',
                    },
                    {
                        description: 'Please specify a maximum image size for desktop display.',
                        force_reload: true,
                        id: 'logo_size',
                        label: 'Logo size',
                        options: [
                            {
                                label: 'Original (as uploaded)',
                                value: 'original',
                            },
                            {
                                label: 'Optimized for theme',
                                value: '250x100',
                            },
                            {
                                label: 'Specify dimensions',
                                value: 'custom',
                            },
                        ],
                        type: 'imageDimension',
                    },
                    {
                        force_reload: true,
                        id: 'logo-position',
                        label: 'Logo position',
                        options: [
                            {
                                label: 'Right',
                                value: 'right',
                            },
                            {
                                label: 'Center',
                                value: 'center',
                            },
                            {
                                label: 'Left',
                                value: 'left',
                            },
                        ],
                        type: 'select',
                    },
                    {
                        force_reload: true,
                        id: 'show_powered_by',
                        label: 'Show Powered By',
                        type: 'checkbox',
                    },
                    {
                        id: 'pdp-sale-price-label',
                        label: 'Product sale price label',
                        type: 'text',
                    },
                ],
            };
            const themeSettings = shallow(
                <ThemeSettings
                    position={{ x: 5, y: 10 }}
                    settingsIndex={1}
                    themeSettings={testItems}
                    settings={settings}
                    updateThemeConfigChange={mockFetch}
                    {...routeProps}
                />
            );

            expect(themeSettings).toMatchSnapshot();
        });
    });

    describe('when handleChange is called', () => {
        describe('when target type is checkbox', () => {
            it('should update themeconfig to new value', () => {
                const testItems = {
                    name: 'Typography & Icons',
                    settings: [
                        {
                            content: 'Logo',
                            type: 'heading',
                        },
                        {
                            id: 'hide_content',
                            label: 'Hide links on webpage',
                            type: 'checkbox',
                        },
                        {
                            id: 'logo-font',
                            label: 'Font family',
                            options: [
                                {
                                    group: 'Oswald',
                                    label: 'Oswald',
                                    value: 'Google_Oswald_400',
                                },
                                {
                                    group: 'Oswald',
                                    label: 'Oswald Light',
                                    value: 'Google_Oswald_300',
                                },
                            ],
                            type: 'font',
                        },
                        {
                            content: 'Headings',
                            type: 'heading',
                        },
                        {
                            id: 'color-textHeading',
                            label: 'Headings text color',
                            type: 'color',
                        },
                        {
                            description: 'Please specify a maximum image size for desktop display.',
                            force_reload: true,
                            id: 'logo_size',
                            label: 'Logo size',
                            options: [
                                {
                                    label: 'Original (as uploaded)',
                                    value: 'original',
                                },
                                {
                                    label: 'Optimized for theme',
                                    value: '250x100',
                                },
                                {
                                    label: 'Specify dimensions',
                                    value: 'custom',
                                },
                            ],
                            type: 'imageDimension',
                        },
                        {
                            force_reload: true,
                            id: 'logo-position',
                            label: 'Logo position',
                            options: [
                                {
                                    label: 'Right',
                                    value: 'right',
                                },
                                {
                                    label: 'Center',
                                    value: 'center',
                                },
                                {
                                    label: 'Left',
                                    value: 'left',
                                },
                            ],
                            type: 'select',
                        },
                        {
                            force_reload: true,
                            id: 'show_powered_by',
                            label: 'Show Powered By',
                            type: 'checkbox',
                        },
                        {
                            id: 'pdp-sale-price-label',
                            label: 'Product sale price label',
                            type: 'text',
                        },
                    ],
                };

                const setting = {
                    id: 'hide_content',
                    label: 'Hide links on webpage',
                    type: 'checkbox',
                };

                let themeSettings: any;

                themeSettings = shallow(
                    <ThemeSettings
                        position={{ x: 5, y: 10 }}
                        settingsIndex={1}
                        themeSettings={testItems}
                        settings={settings}
                        updateThemeConfigChange={mockProp}
                        {...routeProps}
                    />
                );

                const target = {
                    checked: false,
                    type: 'checkbox',
                };

                themeSettings.instance().handleChange(setting)({ target });
                expect(mockProp).toBeCalledWith({ setting, value: target.checked });
            });
        });

        describe('when target is not a checkbox', () => {
            it('should update with the correct value', () => {
                const testItems = {
                    name: 'Typography & Icons',
                    settings: [
                        {
                            content: 'Logo',
                            type: 'heading',
                        },
                        {
                            id: 'hide_content',
                            label: 'Hide links on webpage',
                            type: 'checkbox',
                        },
                    ],
                };

                const setting = {
                    id: 'color-textHeading',
                    label: 'Headings text color',
                    type: 'color',
                };

                let themeSettings: any;

                themeSettings = shallow(
                    <ThemeSettings
                        position={{ x: 5, y: 10 }}
                        settingsIndex={1}
                        themeSettings={testItems}
                        settings={settings}
                        updateThemeConfigChange={mockProp}
                        {...routeProps}
                    />
                );

                const target = {
                    type: 'color',
                    value: '#FF123',
                };

                themeSettings.instance().handleChange(setting)({ target });
                expect(mockProp).toBeCalledWith({ setting, value: target.value });
            });
        });
    });

    describe('getEditor', () => {
        describe('when type equals color', () => {
            it('should return with the correct type', () => {
                const setting = {
                    id: 'testing-color-picker',
                    label: 'Button text color',
                    type: 'color',
                };

                const colorPicker = getEditor(setting, preSetValue, mockProp, mockProp);
                expect(colorPicker).toMatchSnapshot();
            });
        });

        describe('when type equals checkbox', () => {
            it('should return with the correct type', () => {
                const setting = {
                    id: 'testing-checkbox',
                    label: 'Hide links to web pages',
                    type: 'checkbox',
                };

                const checkBox = getEditor(setting, preSetValue, mockProp, mockProp);
                expect(checkBox).toMatchSnapshot();
            });
        });

        describe('when type equals font', () => {
            it('should return with the correct type', () => {
                const setting = {
                    id: 'testing-font',
                    label: 'Logo Font Family',
                    type: 'font',
                };

                const font = getEditor(setting, preSetValue, mockProp, mockProp);
                expect(font).toMatchSnapshot();
            });
        });

        describe('when type equals imageDimension', () => {
            it('should return the correct type', () => {
                const setting = {
                    id: 'test-imageDimension',
                    label: 'Background image',
                    type: 'imageDimension',
                };

                const imageDimension = getEditor(setting, preSetValue, mockProp, mockProp);
                expect(imageDimension).toMatchSnapshot();
            });
        });

        describe('when type equals optimizedCheckout-image', () => {
            it('should return the correct type', () => {
                const setting = {
                    id: 'test-imageUpload',
                    label: 'Upload image',
                    type: 'optimizedCheckout-image',
                };

                const checkoutImageUpload = getEditor(setting, preSetValue, mockProp, mockProp);
                expect(checkoutImageUpload).toMatchSnapshot();
            });
        });

        describe('when type equals select', () => {
            it('should return correct type', () => {
                const setting = {
                    id: 'test-select',
                    label: 'Logo type',
                    type: 'select',
                };

                const selectBox = getEditor(setting, preSetValue, mockProp, mockProp);
                expect(selectBox).toMatchSnapshot();
            });
        });

        describe('when type equals text', () => {
            it('should return correct type', () => {
                const setting = {
                    id: 'test-text',
                    label: '',
                    type: 'text',
                };

                const text = getEditor(setting, preSetValue, mockProp, mockProp);
                expect(text).toMatchSnapshot();
            });
        });

        describe('when type equals heading', () => {
            it('should return correct type', () => {
                const setting = {
                    content: 'HELLO',
                    type: 'heading',
                };
                const heading = getEditor(setting, preSetValue, mockProp, mockProp);
                expect(heading).toMatchSnapshot();
            });
        });

        describe('when type is not listed', () => {
            it('should return null', () => {
                const setting = {
                    content: 'whatever',
                    type: 'not_a_real_type',
                };
                const notRealType = getEditor(setting, preSetValue, mockProp, mockProp);
                expect(notRealType).toEqual(null);
            });
        });
    });
});
