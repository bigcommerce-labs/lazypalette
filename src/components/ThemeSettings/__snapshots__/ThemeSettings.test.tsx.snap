// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ThemeSettings /> getEditor when label has HTML entities (encoded characters) should decode the label 1`] = `
<WithTheme(SelectBox)
  label="<Here> & \\"There's\\" © ®"
  onChange={[Function]}
  options={Array []}
  selected=""
  testId="select.test-labels"
/>
`;

exports[`<ThemeSettings /> getEditor when type equals checkbox should return with the correct type 1`] = `
<CheckboxInput
  checked={false}
  label="Hide links to web pages"
  onChange={
    [MockFunction] {
      "calls": Array [
        Array [
          Object {
            "setting": Object {
              "id": "hide_content",
              "label": "Hide links on webpage",
              "type": "checkbox",
            },
            "value": false,
          },
        ],
        Array [
          Object {
            "setting": Object {
              "id": "color-textHeading",
              "label": "Headings text color",
              "type": "color",
            },
            "value": "#FF123",
          },
        ],
      ],
    }
  }
  testId="checkbox.testing-checkbox"
/>
`;

exports[`<ThemeSettings /> getEditor when type equals color should return with the correct type 1`] = `
<Connect(ColorSetting)
  color="#FFFFF"
  label="Button text color"
  name="testing-color-picker"
  onChange={
    [MockFunction] {
      "calls": Array [
        Array [
          Object {
            "setting": Object {
              "id": "hide_content",
              "label": "Hide links on webpage",
              "type": "checkbox",
            },
            "value": false,
          },
        ],
        Array [
          Object {
            "setting": Object {
              "id": "color-textHeading",
              "label": "Headings text color",
              "type": "color",
            },
            "value": "#FF123",
          },
        ],
      ],
    }
  }
  testId="color.testing-color-picker"
/>
`;

exports[`<ThemeSettings /> getEditor when type equals font should return with the correct type 1`] = `
<WithTheme(SelectBox)
  label="Logo Font Family"
  onChange={
    [MockFunction] {
      "calls": Array [
        Array [
          Object {
            "setting": Object {
              "id": "hide_content",
              "label": "Hide links on webpage",
              "type": "checkbox",
            },
            "value": false,
          },
        ],
        Array [
          Object {
            "setting": Object {
              "id": "color-textHeading",
              "label": "Headings text color",
              "type": "color",
            },
            "value": "#FF123",
          },
        ],
      ],
    }
  }
  options={Array []}
  selected={false}
  testId="font.testing-font"
/>
`;

exports[`<ThemeSettings /> getEditor when type equals heading should return correct type 1`] = `
<styled.div>
  HELLO
</styled.div>
`;

exports[`<ThemeSettings /> getEditor when type equals imageDimension should return the correct type 1`] = `
<ImageSize
  label="Background image"
  onChange={
    [MockFunction] {
      "calls": Array [
        Array [
          Object {
            "setting": Object {
              "id": "hide_content",
              "label": "Hide links on webpage",
              "type": "checkbox",
            },
            "value": false,
          },
        ],
        Array [
          Object {
            "setting": Object {
              "id": "color-textHeading",
              "label": "Headings text color",
              "type": "color",
            },
            "value": "#FF123",
          },
        ],
      ],
    }
  }
  options={Array []}
  selected="20x20"
  testId="imageDimension.test-imageDimension"
/>
`;

exports[`<ThemeSettings /> getEditor when type equals optimizedCheckout-image should return the correct type 1`] = `
<CheckoutImageUpload
  imageURL="hello_there"
  label="Upload image"
  name="test-imageUpload"
  onChange={
    [MockFunction] {
      "calls": Array [
        Array [
          Object {
            "setting": Object {
              "id": "hide_content",
              "label": "Hide links on webpage",
              "type": "checkbox",
            },
            "value": false,
          },
        ],
        Array [
          Object {
            "setting": Object {
              "id": "color-textHeading",
              "label": "Headings text color",
              "type": "color",
            },
            "value": "#FF123",
          },
        ],
      ],
    }
  }
  testId="optimizedCheckout-image.test-imageUpload"
/>
`;

exports[`<ThemeSettings /> getEditor when type equals select should return correct type 1`] = `
<WithTheme(SelectBox)
  label="Logo type"
  onChange={[Function]}
  options={Array []}
  selected="integer:1"
  testId="select.test-select"
/>
`;

exports[`<ThemeSettings /> getEditor when type equals text should return correct type 1`] = `
<InputField
  label=""
  onBlur={
    [MockFunction] {
      "calls": Array [
        Array [
          Object {
            "setting": Object {
              "id": "hide_content",
              "label": "Hide links on webpage",
              "type": "checkbox",
            },
            "value": false,
          },
        ],
        Array [
          Object {
            "setting": Object {
              "id": "color-textHeading",
              "label": "Headings text color",
              "type": "color",
            },
            "value": "#FF123",
          },
        ],
      ],
    }
  }
  onChange={
    [MockFunction] {
      "calls": Array [
        Array [
          Object {
            "setting": Object {
              "id": "hide_content",
              "label": "Hide links on webpage",
              "type": "checkbox",
            },
            "value": false,
          },
        ],
        Array [
          Object {
            "setting": Object {
              "id": "color-textHeading",
              "label": "Headings text color",
              "type": "color",
            },
            "value": "#FF123",
          },
        ],
      ],
    }
  }
  testId="text.test-text"
  value="kermit_the_frog"
/>
`;

exports[`<ThemeSettings /> render() renders the component 1`] = `
<Draggable
  position={
    Object {
      "x": 5,
      "y": 10,
    }
  }
>
  <ExpandableMenu
    title="Typography & Icons"
  >
    <styled.ul>
      <styled.li
        key="0"
      >
        <styled.div>
          Logo
        </styled.div>
      </styled.li>
      <styled.li
        key="1"
      >
        <WithTheme(SelectBox)
          label="Font family"
          onChange={[Function]}
          options={
            Array [
              Object {
                "label": "Oswald",
                "value": "Google_Oswald_400",
              },
              Object {
                "label": "Oswald Light",
                "value": "Google_Oswald_300",
              },
            ]
          }
          testId="font.logo-font"
        />
      </styled.li>
      <styled.li
        key="2"
      >
        <styled.div>
          Headings
        </styled.div>
      </styled.li>
      <styled.li
        key="3"
      >
        <Connect(ColorSetting)
          label="Headings text color"
          name="color-textHeading"
          onChange={[MockFunction]}
          testId="color.color-textHeading"
        />
      </styled.li>
      <styled.li
        key="4"
      >
        <ImageSize
          label="Logo size"
          onChange={[Function]}
          options={
            Array [
              Object {
                "label": "Original (as uploaded)",
                "value": "original",
              },
              Object {
                "label": "Optimized for theme",
                "value": "250x100",
              },
              Object {
                "label": "Specify dimensions",
                "value": "custom",
              },
            ]
          }
          testId="imageDimension.logo_size"
        />
      </styled.li>
      <styled.li
        key="5"
      >
        <WithTheme(SelectBox)
          label="Logo position"
          onChange={[Function]}
          options={
            Array [
              Object {
                "label": "Right",
                "value": "right",
              },
              Object {
                "label": "Center",
                "value": "center",
              },
              Object {
                "label": "Left",
                "value": "left",
              },
            ]
          }
          selected=""
          testId="select.logo-position"
        />
      </styled.li>
      <styled.li
        key="6"
      >
        <CheckboxInput
          label="Show Powered By"
          onChange={[Function]}
          testId="checkbox.show_powered_by"
        />
      </styled.li>
      <styled.li
        key="7"
      >
        <InputField
          label="Product sale price label"
          onBlur={[Function]}
          onChange={[Function]}
          testId="text.pdp-sale-price-label"
        />
      </styled.li>
    </styled.ul>
  </ExpandableMenu>
</Draggable>
`;
